### PR TITLE
Update the project setup guide

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

- The current project setup guide is split into three pages but each of
  them are not very long and thus could be merged into one page.
- We do not distribute `-shade` JARs anymore.

Modifications:

- Merged `setup-*.rst` into `setup.rst`.
- Added a table that lists the major modules and their descriptions.

Result:

- Better-looking documentation
- Up-to-date information in `setup.rst`

### CODE REVIEW

**Comment by trustin** [comment]
Screenshot: https://screenshots.firefox.com/1KMrkDnOMKLQrbiq/null

File: site/src/sphinx/setup.rst [hunk 1]
```diff
@@ -3,8 +3,192 @@
...
+|                                               | See :ref:`server-grpc` and :ref:`client-grpc`.                    |
++-----------------------------------------------+-------------------------------------------------------------------+
+| ``armeria-jetty``                             | Embedded Jetty Servlet container. See :ref:`server-servlet`.      |
++-----------------------------------------------+-------------------------------------------------------------------+
+| ``armeria-kafka``                             | Enables sending access logs to Kafka                              |
++-----------------------------------------------+-------------------------------------------------------------------+
+| ``armeria-logback``                           | Provides Logback ``Appender`` implementation that adds            |
+|                                               | request information. See :ref:`advanced-logging`.                 |
++-----------------------------------------------+-------------------------------------------------------------------+
+| ``armeria-retrofit2``                         | Allows using Armeria instead of OKHTTP as transport layer         |
```
Comment (hyangtack) [inline site/src/sphinx/setup.rst]: nit: OkHttp? (k is lowercase)

File: site/src/sphinx/setup.rst [hunk 2]
```diff
@@ -3,8 +3,192 @@
...
+          <scope>runtime</scope>
+        </dependency>
+      </dependencies>
+    </project>
+
+Using Maven BOM for simpler dependency management
+-------------------------------------------------
+
+You can import ``com.linecorp.armeria:armeria-bom`` into your build rather than specifying Armeria versions in
+more than once place. See `this article <https://www.baeldung.com/spring-maven-bom>`_ to  learn more about what
```
Comment (hyangtack) [inline site/src/sphinx/setup.rst]: nit: `to  learn` -> `to learn` (removing space)

File: site/src/sphinx/setup.rst [hunk 3]
```diff
@@ -3,8 +3,192 @@
...
++-----------------------------------------------+-------------------------------------------------------------------+
+| ``armeria-zipkin``                            | Zipkin distributed tracing support. See :ref:`advanced-zipkin`.   |
++-----------------------------------------------+-------------------------------------------------------------------+
+| ``armeria-zookeeper``                         | ZooKeeper-based service discovery. See :ref:`advanced-zookeeper`. |
++-----------------------------------------------+-------------------------------------------------------------------+
+
+Setting up with Gradle
+----------------------
+
+You might want to use the following  ``build.gradle`` as a starting point when you set up a new project:
```
Comment (hyangtack) [inline site/src/sphinx/setup.rst]: nit: `following  build.gradle` -> `following build.gradle` (removing space)

File: site/src/sphinx/setup.rst [hunk 4]
```diff
@@ -3,8 +3,192 @@
...
+
+        // Logging
+        runtime 'ch.qos.logback:logback-classic:\ |ch.qos.logback:logback-classic:version|\ '
+        runtime 'org.slf4j:log4j-over-slf4j:\ |org.slf4j:log4j-over-slf4j:version|\ '
+    }
+
+Setting up with Maven
+---------------------
+
+You might want to use the following  ``pom.xml`` as a starting point when you set up a new project:
```
Comment (hyangtack) [inline site/src/sphinx/setup.rst]: ditto

File: site/src/sphinx/setup.rst (hunk 1 above)
Comment (trustin) [reply site/src/sphinx/setup.rst]: Fixed

File: site/src/sphinx/setup.rst (hunk 2 above)
Comment (trustin) [reply site/src/sphinx/setup.rst]: Thanks! Fixed

File: site/src/sphinx/setup.rst (hunk 3 above)
Comment (trustin) [reply site/src/sphinx/setup.rst]: Fixed

File: site/src/sphinx/setup.rst (hunk 4 above)
Comment (trustin) [reply site/src/sphinx/setup.rst]: Fixed

**Pull request merged**